src/xopen/__init__.py

```diff
@@ -202,7 +206,10 @@ def _open_process(
     ) -> Popen:
         program_args: List[str] = self._program_args[:]  # prevent list aliasing
         if threads != 0 and self._threads_flag is not None:
-            program_args += [self._threads_flag, str(threads)]
+            if self._space:
+                program_args += [self._threads_flag, str(threads)]
+            else:
+                program_args += [f"{self._threads_flag}{threads}"]
```

Comment:
This if else clause is making the program more complex than it should be. Currently the only other program accepting threads is `pigz` and it also parses `-p4` correctly.

So it is best IMO to remove all the `_space` stuff and simply change the line on 205 with the line on 212. No if clauses needed.

Reply:
For clarification, just keep the old code the same, but replace `program_args += [self._threads_flag, str(threads)]` with `program_args += [f"{self._threads_flag}{threads}"]`. That solves the space problem and will not cause problems for pigz. Also this makes the `space` parameter that was added redundant. Doing it this way keeps it simpler.

Reply:
Ah, of course! Adjusted.